tbdex/quote/quote.go

```diff
@@ -0,0 +1,252 @@
+package quote
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/TBD54566975/tbdex-go/tbdex"
+	"github.com/tbd54566975/web5-go/dids/did"
+	"go.jetpack.io/typeid"
+)
+
+// Kind identifies this message kind
+const Kind = "quote"
+
+// Quote represents a quote message within the exchange.
+type Quote struct {
+	MessageMetadata tbdex.MessageMetadata `json:"metadata"`
+	Data            Data                  `json:"data"`
+	Signature       string                `json:"signature"`
```

Comment:
we should include the `omitempty` tag on all required fields. This will ensure that the field is _only_ present in the JSON payload when the field is not the zero value. Otherwise, callers end up sending a request with a field that has the zero value, which probably isn't desired. 

Suggesting this _only_ for the files you've added in this PR. will address in rest of this repo in a subsequent PR as i know you're trying to keep PRs specific to what you intended to tackle. created issue to track: https://github.com/TBD54566975/tbdex-go/issues/28

Reply:
Ah ok, I had been using `omitempty` _only_ on optional fields lol

Reply:
good to know!

Reply:
but a question - i had been looking at presence / absence of `omitempty` to deduce which one's optional and which one's required - so i guess that heuristic is now gone? is there another way i could use to tell which fields are required and which are optional?